.github/wiki/Tests.md

```diff
@@ -0,0 +1,135 @@
+
+[buck-install]: https://buckbuild.com/setup/install.html
+
+To run the Android unit tests, invoke the following script from the root of your React Native checkout:
+
+```bash
+./scripts/run-android-local-unit-tests.sh
+```
+
+The Android integration tests, on the other hand, need additional setup. We recommend going through the instructions to [set up your environment for building React Native from source](http://facebook.github.io/react-native/docs/building-from-source#prerequisites).
+
+Once you've done that, you can start the Android emulator using:
+
+```bash
+./scripts/run-android-emulator.sh
+```
+
+Then, run the Android integration tests:
+
+```bash
+./scripts/run-android-local-integration-tests.sh
+```
+
+### End-to-end Tests
+
+Finally, make sure end-to-end tests run successfully by executing the following script:
+
+```bash
+./scripts/test-manual-e2e.sh
+```
+
+End-to-end tests written in [Detox](https://github.com/wix/Detox) confirm that React Native components and APIs function correctly in the context of a running app. They run the RNTester app in the simulator and simulate a user interacting with the app.
+
+You can run Detox end-to-end tests locally by [installing the Detox CLI](https://github.com/wix/Detox/blob/master/docs/Introduction.GettingStarted.md#step-1-install-dependencies) on macOS, then running the following in the command line:
+
+```bash
+npm run build-ios-e2e
+npm run test-ios-e2e
+```
+
+If you work on a component or API that isn't convered by a Detox test, please consider adding one. Detox tests are stored under [`RNTester/e2e/__tests__`](https://github.com/facebook/react-native/tree/master/RNTester/e2e/__tests__).
+
+### Continuous Testing
+
+We use [Appveyor][config-appveyor] and [Circle CI][config-circleci] to automatically run our open source tests. Appveyor and Circle CI will run these tests whenever a commit is added to a pull request, as a way to help maintainers understand whether a code change introduces a regression. The tests also run on commits to the master and `*-stable` branches in order to keep track of the health of these branches.
+
+[config-appveyor]: https://github.com/facebook/react-native/blob/master/.appveyor/config.yml
+[config-circleci]: https://github.com/facebook/react-native/blob/master/.circleci/config.yml
+
+There's another set of tests that run within Facebook's internal test infrastructure. Some of these tests are integration tests defined by internal consumers of React Native (e.g. unit tests for a React Native surface in the Facebook app). These tests run on every commit to the copy of React Native hosted on Facebook's source control. They also run when a pull request is imported to Facebook's source control. As it happens, should one of these tests fail, you'll need someone at Facebook to take a look. Since pull requests can only be imported by Facebook employees, whoever imported the pull request should be able to facilitate any details.
```

Comment:
```suggestion
There's another set of tests that run within Facebook's internal test infrastructure. Some of these tests are integration tests defined by internal consumers of React Native (e.g. unit tests for a React Native surface in the Facebook app). These tests run on every commit to the copy of React Native hosted on Facebook's source control. They also run when a pull request is imported to Facebook's source control.

If one of these tests fail, you'll need someone at Facebook to take a look. Since pull requests can only be imported by Facebook employees, whoever imported the pull request should be able to facilitate any details.
```